sql/analyzer/validate_create_table.go

```diff
@@ -145,7 +180,9 @@ func validateAddColumn(initialSch sql.Schema, schema sql.Schema, ac *plan.AddCol
 	}
 
 	// None of the checks we do concern ordering, so we don't need to worry about it here
-	newSch := append(table.Schema(), ac.Column())
+	newCol := copyColumn(ac.Column())
```

Comment:
Would add Column.Copy() in the sql package, generally useful for things like this